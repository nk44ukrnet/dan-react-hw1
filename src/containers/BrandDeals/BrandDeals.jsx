import React from 'react';
import Container from "../Container/Container.jsx";
import Brand1 from '/public/brand1.png';
import Brand2 from '/public/brand2.png';
import Brand3 from '/public/brand3.png';
import Brand4 from '/public/brand4.png';
import Brand5 from '/public/brand5.png';
import './BrandDeals.scss';

export default function BrandDeals() {
    // const Brand1 = require('/public/brand1.png');
    // const Brand2 = require('/public/brand2.png');
    // const Brand3 = require('/public/brand3.png');
    // const Brand4 = require('/public/brand4.png');
    // const Brand5 = require('/public/brand5.png');
    return (
        <div className="brand-deals">
            <Container>
                <div className="brand-deals__inner">
                    <h2 className="brand-deals__heading">Top Brands Deal</h2>
                    <p className="brand-deals__subheading">Up To <span>60%</span> off on brands</p>
                    <div className="brand-deals__images">
                        <img src={Brand1} alt="brand logo" loading="lazy"/>
                        <img src={Brand2} alt="brand logo" loading="lazy"/>
                        <img src={Brand3} alt="brand logo" loading="lazy"/>
                        <img src={Brand4} alt="brand logo" loading="lazy"/>
                        <img src={Brand5} alt="brand logo" loading="lazy"/>
                    </div>
                </div>
            </Container>
        </div>
    );
};