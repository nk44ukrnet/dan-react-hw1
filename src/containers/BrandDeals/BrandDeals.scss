.brand-deals {
  margin-block: 20px;

  &__inner {
    border: 1.07px solid #323232;
    border-radius: 12px;
    background-color: #323232;
    padding: 40px 20px 60px;
    box-sizing: border-box;
    color: #fff;
    text-align: center;
  }

  &__heading {
    font-weight: 800;
    font-size: 50px;
    text-align: center;
    margin: 0;
    margin-bottom: 25px;
  }

  &__subheading {
    font-weight: 400;
    font-size: 22px;
    text-align: center;
    margin: 0;
    margin-bottom: 67px;

    span {
      font-weight: 700;
      color: #fbd103;
    }
  }

  &__images {
    display: flex;
    flex-wrap: wrap;
    gap: 25px;
    justify-content: center;
  }
}