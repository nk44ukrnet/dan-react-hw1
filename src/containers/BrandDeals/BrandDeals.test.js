import React from 'react';
import { render } from '@testing-library/react';
import BrandDeals from './BrandDeals.jsx';

jest.mock('/public/brand1.png', () => 'brand1.png');
jest.mock('/public/brand2.png', () => 'brand2.png');
jest.mock('/public/brand3.png', () => 'brand3.png');
jest.mock('/public/brand4.png', () => 'brand4.png');

describe('BrandDeals Component', () => {
    it('should match the snapshot', () => {
        const brandDeals = render(<BrandDeals />);
        expect(brandDeals).toMatchSnapshot();
    });
});