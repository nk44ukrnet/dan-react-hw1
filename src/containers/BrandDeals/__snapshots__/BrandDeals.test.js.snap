// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandDeals Component should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="brand-deals"
      >
        <div
          class="container"
        >
          <div
            class="brand-deals__inner"
          >
            <h2
              class="brand-deals__heading"
            >
              Top Brands Deal
            </h2>
            <p
              class="brand-deals__subheading"
            >
              Up To 
              <span>
                60%
              </span>
               off on brands
            </p>
            <div
              class="brand-deals__images"
            >
              <img
                alt="brand logo"
                loading="lazy"
                src="brand4.png"
              />
              <img
                alt="brand logo"
                loading="lazy"
                src="brand4.png"
              />
              <img
                alt="brand logo"
                loading="lazy"
                src="brand4.png"
              />
              <img
                alt="brand logo"
                loading="lazy"
                src="brand4.png"
              />
              <img
                alt="brand logo"
                loading="lazy"
                src="brand4.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="brand-deals"
    >
      <div
        class="container"
      >
        <div
          class="brand-deals__inner"
        >
          <h2
            class="brand-deals__heading"
          >
            Top Brands Deal
          </h2>
          <p
            class="brand-deals__subheading"
          >
            Up To 
            <span>
              60%
            </span>
             off on brands
          </p>
          <div
            class="brand-deals__images"
          >
            <img
              alt="brand logo"
              loading="lazy"
              src="brand4.png"
            />
            <img
              alt="brand logo"
              loading="lazy"
              src="brand4.png"
            />
            <img
              alt="brand logo"
              loading="lazy"
              src="brand4.png"
            />
            <img
              alt="brand logo"
              loading="lazy"
              src="brand4.png"
            />
            <img
              alt="brand logo"
              loading="lazy"
              src="brand4.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
